frontend/app/src/pages/home/join.tsx

```diff
@@ -1,11 +1,50 @@
+import { useState } from "react";
+import { useNavigate } from "react-router-dom";
+import { io } from "socket.io-client";
+
 import { Page } from "@/components";
+import { MATCHING_EVENTS } from "@/constants/matching";
+import { ROUTE } from "@/constants/route";
 import { FindingMatchCard, SelectPreferencesCard } from "@/features/matching";
+import { env } from "@/lib/env";
+import { Matching, Preferences, matchingSchema } from "@/types/matching";
 
 function JoinPage() {
+  const navigate = useNavigate();
+  const socket = io(env.VITE_MATCHING_SERVICE_URL);
+  const [isWaitingForMatch, setIsWaitingForMatch] = useState<boolean>(false);
```

Comment:
```suggestion
  const [isWaitingForMatch, setIsWaitingForMatch] = useState(false);
```

don't need explicit typing for types that can be inferred

Reply:
resolved in 343093b18dcc65d968691ffbe27f35441ff77a2f